src/schemas.jl

```diff
@@ -638,6 +674,7 @@ function _generate_record_type_definitions(schema_version::SchemaVersion, record
         end
         outer_constructor_definitions = quote
             $outer_constructor_definitions
+            $R{$(type_param_names...)}(::NamedTuple) where {$(type_param_names...)} = $R{$(type_param_names...)}($Legolas.DontCheckArgs())
             $R{$(type_param_names...)}(row) where {$(type_param_names...)} = $R{$(type_param_names...)}(; $(kwargs_from_row...))
```

Comment:
I don't undersatnd what's happening here.  Doesn't this change the existing behavior (the case when `row` is a `NamedTuple`)?  if it _doesn't_ change the behavior, it at the very least needs a comment explaining why this is reasonable.